docs/pipelines.md

```diff
@@ -479,6 +479,16 @@ params:
     value: "$(tasks.checkout-source.results.commit)"
 ```
 
+In the snippet below, a `WhenExpression` is provided its value from the `status` `Result` emitted by the
```

Comment:
im not sure i'd include an example where the result is called `status` - in #3187 the person who opened the issue thought they could check the status of a previous Task, but we don't support that - i think calling a result "status" might further muddy that